webinar-series/components/slides.md

```diff
@@ -1,10 +1,761 @@
+</body>
+```
+
+You can start to imagine an web app as a composable set of custom 'HTML Primitives' like this!
+
+---
+
+### The Evolution of Angular Components
+
+* This way of using directives got very popular
+* Angular 1.5 introduced first-class support for the pattern
+* The new `.component` function:
+
+```javascript
+angular.module('ngcourse')
+  .component('ngcHelloComponent', {
+    bindings: { name: '=' },
+    template: '<span>Hello, {{ ctrl.name }}.</span>',
+    controller: function MyComponentCtrl() {
+      // component logic here.
+    }
+  });
+```
+
+---
+
+### The Evolution of Angular Components
+
+In Angular 2 it's the main way to build UI:
+
+```typescript
+import {Component} from 'angular2/core';
+
+@Component({
+  selector: 'ngc-hello-component',
+  template: '<p class="hello">Hello, {{name}}</p>',
+  styles: [ '.hello { color: purple; }' ]
+})
+export class HelloComponent {
+  name: string;
+  constructor() {
+    this.name = 'World';
+  }
+}
+```
+
+* Structure as an HTML `template`
+* 'DOM name' using the `selector` property
+* Styling using the `styles` property
+* Behaviour using methods and properties on the `class`
+
+.bottom-comment[<a target="\_blank" href="http://plnkr.co/edit/Z7KNUe8UcooxO6dfAayL?p=preview">See it live</a>]
+
+---
+
+### Architectural Considerations
+
+* Angular requires one top-level (bootstrap) component
+* Other components can be very granular
+* Most components are presentational
+
+  * 'Dumb' components
+  * Very little logic
+  * Very little state
+  * Maybe even just markup and CSS
+  * Highly re-usable
+
+* Motivation: Atomic Design (http://bradfrost.com/blog/post/atomic-web-design/)
+  * Comes from the design World
+  * Intersection of UI development and graphic design/branding
+  * Encapsulation of presentation instead of behaviour or state
+
+* We're all converging on components!
+
+---
+
+### Atomic Design
+
+> We’re not designing pages, we’re designing systems of components.
+>
+> .attribution[—Stephen Hay]
+
+* We build up our UIs from very granular components or 'atoms'
+* They are highly re-usable
+* Incorporate design and branding at the lowest widget level:
+
+.center[![atoms](../../img/atoms.png)]
+
+---
+
+### Atomic Design
+
+* These atoms form a 'living style guide'
+* Are combined into more useful 'molecules', 'organisms', and finally 'pages':
+
+.center[![molecule](../../img/molecule.png)]
+
+
+.flex.justify-between.bold.rangle-purple.caps[
+  .block[More Reusable]
+  .block[Less Reusable]]
+
+.img-max-width[![left-bar](../../img/bar-max-left.svg)]
+
+.flex.justify-between.bold.rangle-purple.caps[
+  .block.border.rounded.p1[Atoms]
+  .block.border.rounded.p1[Molecules]
+  .block.border.rounded.p1[Organisms]
+  .block.border.rounded.p1[Pages]]
+
+.img-max-width[![right-bar](../../img/bar-max-right.svg)]
+
+.flex.justify-between.bold.rangle-purple.caps[
+  .block[Less Complex]
+  .block[More Complex]]
+
+---
+
+### Smart vs. Dumb Components
```

Comment:
Have you considered using the terms Presentational & Container components. I've been seeing those everywhere lately – even the ngCourse uses it in the redux section. IMO it sounds better.